subsys/bootloader/provision/provision_flash.c

```diff
@@ -24,6 +24,17 @@ typedef struct {
 	} key_data[1];
 } provision_flash_t;
 
+struct monotonic_counter {
+	u16_t description;
+	u16_t num_counter_slots;
+	u16_t counter_slots[1];
+};
+struct monotonic_counters {
+	u32_t num_counters;
+	struct monotonic_counter counters[1];
+};
+
+#define DESC_VERSION_COUNTER 1
```

Comment:
It is not clear to me what this is used for, should we have a comment here?

Reply:
Adding `/* Counter description value for firmware version. */`